packages/react-sdk/src/state/synchronizedDocumentImpl.ts

```diff
@@ -158,6 +160,23 @@ export class SynchronizedDocumentImpl<T extends Record<string, unknown>>
             } catch (e) {
               this.logger.error('Could not store snapshot for', documentId, e);
             }
+
+            const previewsEnabled =
+              getEnvironment('REACT_APP_PREVIEWS', 'false') === 'true';
+            if (previewsEnabled) {
+              // we also take the opportunity to store a preview of the document
+              const parentElement =
+                window.document.querySelector('#document-preview');
+              const svgElement = parentElement?.querySelector('svg') as unknown;
+              try {
+                const documentPreview = await svg2preview(
+                  svgElement as HTMLElement,
+                );
+                await this.createDocumentPreview(documentId, documentPreview);
```

Comment:
For some reason, this gets never resolved. However it sends the preview.
But after the first snapshot / preview it stops working.

Reply:
I am observing this behaviour indeed but it's not consistently after the first snapshot. For me, sometimes it stops saving previews after many have been already done...

Reply:
Fixed. The permissions to retrieve the state events was not listed.